src/components/TicketDetailModal/TicketDetailModal.tsx

```diff
@@ -0,0 +1,308 @@
+import _EditableTicket from "../EditableTicket";
+import IconButton from "../IconButton";
+
+export interface TicketDetailModalProps {
+  isOpen: boolean;
+  ticketId: number | null;
+  onClose?: () => void;
+}
+
+export default function TicketDetailModal({ isOpen, ticketId, onClose }: TicketDetailModalProps): JSX.Element {
+  const project = useTypeSafeContext(ProjectContext);
+  const { data: ticketResponse } = useFetch<TicketDefenition>(GET_TICKET_BY_ID(project.id, ticketId!));
+
+  return (
+    <Dialog isOpen={isOpen} onClose={onClose} disableCloseButton>
+      <Card>
+        <Container>
+          <CloseIconButton onClick={onClose}>
+            <Close />
+          </CloseIconButton>
+
+          <Headline>
+            <Title />
+
+            <TicketTitleEditableLabel value={ticketResponse?.name} />
+          </Headline>
+
+          <Content>
+            <Main>
+              <Paragraph>
+                <ParagraphTitle>
+                  <Notes />
+                  <MainTitle>Description</MainTitle>
+                </ParagraphTitle>
+
+                <DescriptionForm>
+                  <DescriptionFormContainer>
+                    <EditableDescription
+                      placeholder="Add a more detailed description…"
+                      value={ticketResponse?.description}
+                      readOnly
+                    />
+
+                    <DescriptionButtons>
+                      <DescriptionSaveButton title="Save" />
+                      <Button title="Cancel" />
+
+                      <Button title="Formatting help" />
+                    </DescriptionButtons>
+                  </DescriptionFormContainer>
+                </DescriptionForm>
+              </Paragraph>
+
+              <Paragraph>
+                <ParagraphTitle>
+                  <FormatListBulleted />
+                  <MainTitle>Activity</MainTitle>
+
+                  <Button title={"Show details"} />
+                </ParagraphTitle>
+
+                <Comments>
+                  <Comment>
+                    <Avatar
+                      alt="Mr Pug"
+                      src="https://ichef.bbci.co.uk/news/976/cpsprodpb/17638/production/_124800859_gettyimages-817514614.jpg"
+                      size="S"
+                    />
+
+                    <EditableTicket placeholder="Write a comment..." />
+                  </Comment>
+                </Comments>
+              </Paragraph>
+            </Main>
+
+            <Side>
+              <SideTitle>Add to card</SideTitle>
+              <Button title="Members" icon={<Person />} />
+              <Button title="Labels" icon={<Label />} />
+              <Button title="Checklist" icon={<Check />} />
+              <Button title="Dates" icon={<AccessTime />} />
+              <Button title="Attachment" icon={<AttachFile />} />
+              <Button title="Cover" icon={<Style />} />
+              <Button title="Custom Fields" icon={<CropLandscape />} />
```

Comment:
You might be able to create an object that contains icon, title, and callback function(if you need) and iterate through with a map function.

```typescript
type SideButton = { title: string, icon: JSX.Element(could be better type), onPress?: () => void};

const ADD_TO_CARD_BUTTONS: SideButton[] = [
  {title: "Members" icon: <Person />},
  {title:"Labels" icon: <Label />}
  ...
]

<>
  {ADD_TO_CARD_BUTTONS.map((buttonItem) => <Button title={buttonItem.title} icon={buttonItem.icon} onPress={buttonItem.onPress} />)}
</>
```